### PR TITLE
[js.spec] Remove unnecessary Predicate interface and add missing params

### PR DESCRIPTION
**Pull request by mattbishop**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/prayerslayer/js.spec/blob/master/lib/spec/spec.js#L17, https://github.com/prayerslayer/js.spec/blob/master/lib/predicates.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by mattbishop** [comment]
@arichiardi would you please have a look at this PR?

File: types/js.spec/index.d.ts [hunk 1]
```diff
@@ -27,18 +27,29 @@ export interface Spec {
...
+     * @returns the human-readable description of this spec
+     */
+    toString(): string;
 }
 
 /**
- * A Spec with a boolean conform function. Test the value and return true if it conforms.
+ * Predicate function definition to describe a predicate function.
  */
-export interface Predicate extends Spec {
```
Comment (arichiardi) [inline types/js.spec/index.d.ts]: Removing this is the correct thing to do.

File: types/js.spec/index.d.ts [hunk 2]
```diff
@@ -27,18 +27,29 @@ export interface Spec {
...
-    (value: any): boolean;
-}
+export type PredFn = (value: any) => boolean;
```
Comment (arichiardi) [inline types/js.spec/index.d.ts]: :+1:

**Review by arichiardi — APPROVED** [review]
LGTM

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**